SimG4Core/SensitiveDetector/src/SensitiveDetector.cc

```diff
@@ -119,13 +118,9 @@ void SensitiveDetector::NaNTrap( G4Step* aStep )
          edm::isNotFinite(xyz[0]) != 0 || edm::isNotFinite(xyz[1]) != 0 || 
 	 edm::isNotFinite(xyz[2]) != 0 )
     {
-       std::cout << " NaN detected in volume " << NameOfVol << std::endl ;
-       throw SimG4Exception( "SimG4CoreSensitiveDetector: Corrupted Event - NaN detected (3-momentum)" ) ;
+      throw SimG4Exception( "SimG4CoreSensitiveDetector: Corrupted Event - NaN detected (3-momentum) in volume " + NameOfVol);
```

Comment:
@davidlange6 , because it is within Geant4 event loop and at that moment it is useful to apply SimG4Exception, I change this file in order to remove  cout.